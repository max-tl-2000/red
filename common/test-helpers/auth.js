/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import moment from 'moment'; // eslint-disable-line red/no-moment
import { expect } from 'chai';

export const verifyTokenExpiration = (decodedToken, expectedTokenExpiration) => {
  const iat = moment.unix(decodedToken.iat);
  const exp = moment.unix(decodedToken.exp);

  const result = exp.diff(iat, expectedTokenExpiration.substring(expectedTokenExpiration.length - 1));
  const expectedValue = expectedTokenExpiration.substring(0, expectedTokenExpiration.length - 1);
  expect(result).to.equal(+expectedValue);
};
