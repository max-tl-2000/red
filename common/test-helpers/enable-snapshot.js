/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import chai from 'chai';
import chaiJestSnapshot from 'chai-jest-snapshot';

const enableSnapshot = () => {
  if (!global.before) return;
  chai.use(chaiJestSnapshot);
  before(() => {
    chaiJestSnapshot.resetSnapshotRegistry();
  });

  beforeEach(function beforeEach() {
    const { currentTest } = this;
    chaiJestSnapshot.setFilename(`${currentTest.file}.chai-snap`);
    chaiJestSnapshot.setTestName(currentTest.fullTitle());
  });
};

enableSnapshot();

export { editUnpredictable } from '../helpers/snapshot-helper';
