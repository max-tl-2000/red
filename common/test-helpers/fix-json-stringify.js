/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

const stringify = JSON.stringify;

before(() => {
  JSON.stringify = obj => {
    const seen = [];

    return stringify(obj, (key, val) => {
      if (typeof val === 'object') {
        if (seen.indexOf(val) >= 0) {
          return;
        }
        seen.push(val);
      }
      return val; // eslint-disable-line
    });
  };
});

after(() => {
  JSON.stringify = stringify;
});
