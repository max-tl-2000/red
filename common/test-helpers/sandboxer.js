/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

let _sinon;
let sandboxes = [];

function create() {
  // this is for when this module is used in karma
  _sinon = _sinon || global.sinon;
  const sandbox = _sinon.sandbox.create();
  sandboxes.push(sandbox);

  return sandbox;
}

function restore() {
  sandboxes.forEach(sandbox => sandbox.restore());
  sandboxes = [];
}

function setSinon(sinon) {
  _sinon = sinon;
}

export { create, restore, setSinon };
