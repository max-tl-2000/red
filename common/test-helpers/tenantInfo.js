/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

export const TEST_TENANT_ID = 'af20764d-1e0d-41a5-be54-3766215f89a4';
