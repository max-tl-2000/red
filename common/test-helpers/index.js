/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import chai from 'chai';
import sinon from 'sinon';
import sinonChai from 'sinon-chai';
import { unmountComponentAtNode } from 'react-dom';
import when from './when';
import and from './and';
import * as overrider from './overrider';
import * as sandboxer from './sandboxer';
import sleep from '../helpers/sleep';

sandboxer.setSinon(sinon);

const expect = chai.expect;
const assert = chai.assert;

chai.use(sinonChai);

let div;

if (global.document) {
  beforeEach(() => {
    div = document.createElement('div');
    div.setAttribute('id', 'sandbox');

    document.body.appendChild(div);
  });
}

afterEach(() => {
  if (global.document && div) {
    unmountComponentAtNode(div);
    document.body.removeChild(div);
    div = null;
  }

  overrider.restore();
  sandboxer.restore();
});

const getSandBoxDiv = () => div;

// helper method to trigger a real click event in the browser
// jquery.fn.click is not the same as this as the event need
// to also generate the capture phase
const triggerMouseEvent = (ele, evt) => {
  const mEvent = document.createEvent('MouseEvent');
  // interface
  // event.initMouseEvent(type, canBubble, cancelable, view,
  //                    detail, screenX, screenY, clientX, clientY,
  //                    ctrlKey, altKey, shiftKey, metaKey,
  //                    button, relatedTarget);
  mEvent.initMouseEvent(evt, true, true, window, 0, 0, 0, 0, 0, false, false, false, false, 0, null);

  ele.dispatchEvent(mEvent);
};

export { getSandBoxDiv, triggerMouseEvent, overrider, expect, assert, sinon, when, sandboxer, and, sleep };
