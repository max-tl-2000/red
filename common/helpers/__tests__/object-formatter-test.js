/*
 * Copyright (c) 2022 Reva Technology Inc., all rights reserved.
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Licensed under the Elastic License 2.0; you may not use this file except
 * in compliance with the Elastic License 2.0.
 */

import { readFileSync } from 'fs'; // had to use readFileSync, for some reason xfs.read was not working
import path from 'path';
import expand from 'glob-expand';
import { when } from '../../test-helpers';
import objectFormatter from '../object-formatter';
import tryParse from '../try-parse';

/**
 * reads the sample json files from the provided glob
 *
 * @param      {string}  glob   the glob to match files from
 * @return     {Array}          array of objects with the following structure:
 * {
 *   name,         // the name of the record key. This is inferred from the first
 *                 // part of the filename before the -- mark
 *   rec,          // the parsed JSON entity to be passed to objectFormatter
 * }
 */
const readSampleRecords = () => {
  const pattern = path.join(__dirname, './fixtures/objs/**/*.json');
  const recordFiles = expand(pattern);
  return recordFiles.map(f => ({
    name: path.basename(f, '.json').split('--')[0],
    rec: tryParse(readFileSync(f, { encoding: 'utf8' }), {}),
  }));
};

describe('objectFormatter.format', () => {
  when('format receives an object like:', () => {
    const recordExamples = readSampleRecords();

    recordExamples.forEach(entry => {
      it(`should format ${entry.name} as expected`, async () => {
        const result = objectFormatter.format(entry.name, entry.rec);
        expect(result).toMatchSnapshot(entry.name);
      });
    });
  });
});
