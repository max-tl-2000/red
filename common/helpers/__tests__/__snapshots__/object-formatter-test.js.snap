// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`objectFormatter.format when format receives an object like: should format err as expected: err 1`] = `
"[31mError: Pool was destroyed
    at processImmediate [as _immediateCallback] (timers.js:383:17)[39m
"
`;

exports[`objectFormatter.format when format receives an object like: should format error as expected: error 1`] = `
"[31mError: Pool was destroyed
    at processImmediate [as _immediateCallback] (timers.js:383:17)[39m
"
`;

exports[`objectFormatter.format when format receives an object like: should format header as expected: header 1`] = `
"HTTP/1.1 200 OK
X-Request-Id: 78525e46-287e-4103-ad6c-4a765f44d410
Content-Type: application/json; charset=utf-8
Content-Length: 388
ETag: W/\\"184-QQ9FH8wt/CPcyjrEx9bPpw\\"
Date: Thu, 30 Jun 2016 08:55:41 GMT
Connection: close"
`;

exports[`objectFormatter.format when format receives an object like: should format headers as expected: headers 1`] = `
"host: 127.0.0.1:65439
accept-encoding: gzip, deflate
user-agent: node-superagent/1.8.3
connection: close
authorization: Bearer eyJ0eXAiOiJKV..."
`;

exports[`objectFormatter.format when format receives an object like: should format req as expected: req 1`] = `
"[36mâ‡ â‡  [31150b4a-a680-430f-8a30-095cac9c1506] â‡ â‡ 
[39m[90m  GET /users
[39m[90m  headers: 
[39m[90m    cookie: i18next=en-US
    accept-language: en-US,en;q=0.8,es;q=0.6
    accept-encoding: gzip, deflate, sdch
    referer: http://red.localhost.com:3000/
    accept: */*
    user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.106 Safari/537.36
    cache-control: no-cache
    pragma: no-cache
    connection: close
    host: red.localhost.com:3000
    authorization: Bearer eyJ0eXAiOiJKV...
[39m[90m  body: 
{ some: 'parameter', was: 'provided' }
[39m[36mâ‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ 
[39m"
`;

exports[`objectFormatter.format when format receives an object like: should format req as expected: req 2`] = `
"[36mâ‡ â‡  [31150b4a-a680-430f-8a30-095cac9c1506] â‡ â‡ 
[39m[90m  GET /users
[39m[90m  headers: 
[39m[90m    cookie: i18next=en-US
    accept-language: en-US,en;q=0.8,es;q=0.6
    accept-encoding: gzip, deflate, sdch
    referer: http://red.localhost.com:3000/
    accept: */*
    user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.106 Safari/537.36
    cache-control: no-cache
    pragma: no-cache
    connection: close
    host: red.localhost.com:3000
    authorization: Bearer eyJ0eXAiOiJKV...
[39m[36mâ‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ â‡ 
[39m"
`;

exports[`objectFormatter.format when format receives an object like: should format res as expected: res 1`] = `
"[31mâ‡¢â‡¢ [a0b20b83-9364-4864-8e55-7d867afb5e13] â‡¢â‡¢
[39m[90m  status: [31m404[90m
[39m[90m  duration (ms): 68
[39m[90m  headers: 
[39m[90m    HTTP/1.1 404 Not Found
    x-powered-by: Express
    X-Request-Id: a0b20b83-9364-4864-8e55-7d867afb5e13
    x-content-type-options: nosniff
    content-type: text/html; charset=utf-8
    content-length: 54
    date: Thu, 30 Jun 2016 15:17:25 GMT
    connection: close
[39m[31mâ‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢
[39m"
`;

exports[`objectFormatter.format when format receives an object like: should format res as expected: res 2`] = `
"[33mâ‡¢â‡¢ [31150b4a-a680-430f-8a30-095cac9c1506] â‡¢â‡¢
[39m[90m  status: [31m500[90m
[39m[90m  duration (ms): 19
[39m[33mâ‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢
[39m"
`;

exports[`objectFormatter.format when format receives an object like: should format res as expected: res 3`] = `
"[33mâ‡¢â‡¢ [31150b4a-a680-430f-8a30-095cac9c1506] â‡¢â‡¢
[39m[90m  status: [32m200[90m
[39m[90m  duration (ms): 19
[39m[90m  headers: 
[39m[90m    HTTP/1.1 200 OK
    X-Request-Id: 31150b4a-a680-430f-8a30-095cac9c1506
    Content-Type: application/json; charset=utf-8
    Content-Length: 1972
    ETag: W/\\"7b4-gQ9mq+61b//bdnqoDPDz4A\\"
    Date: Sat, 02 Jul 2016 03:15:24 GMT
    Connection: close
[39m[33mâ‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢â‡¢
[39m"
`;

exports[`objectFormatter.format when format receives an object like: should format src as expected: src 1`] = `"[90m  /Users/roy/royws/red/server/server.js:190[39m"`;

exports[`objectFormatter.format when format receives an object like: should format unknown as expected: unknown 1`] = `"unknown: { somekey: 'object', boolean: true, arr: [ 1, 2, 3 ] }"`;
